### PR TITLE
Hook up linker to all generated /bin:foo targets

### PR DESCRIPTION
**Pull request by alexeagle**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature (please, look at the "Scope of the project" section in the README.md file)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

**Comment by alexeagle** [comment]
This was peeled out of #1430 to reduce scope there making it easier to land

**Comment by alexeagle** [comment]
finally getting back to this now (I installed a windows boot camp so I can really debug, once I get things like git in cmd.exe sorted...)

@alan-agius4 you had some recommendation for fixing this (.exe or non-interactive shell IIRC) but I think it's lost to the bazel freebie slack dropping messages

File: packages/node-patches/src/subprocess.ts [hunk 1]
```diff
@@ -47,7 +47,8 @@ fi
   }
 
   // fix execPath so folks use the proxy node
-  process.argv[0] = process.execPath = path.join(nodeDir, 'node');
+  process.argv[0] = process.execPath =
```
Comment (alan-agius4) [inline packages/node-patches/src/subprocess.ts:50]: In Windows, this will work only if `shell` is enabled when third parties `fork`or `spawn` a process.

**Comment by alan-agius4** [comment]
@alexeagle I did have the same issue this week when I tried to look up an old convo.

So in Windows you cannot `spawn` or `fork` a `.bat` or `.cmd` files unless the shell option is enabled. (Which is not by default). The problem is that `fork` doesn't have a `shell` option and when provided it is ignored. See https://nodejs.org/api/child_process.html#child_process_child_process_fork_modulepath_args_options

My idea to get around this is to leverage the fact that when a child process is spawned/forked both `env` and `execArgv` are inherited from the parent process. This is important because this means that we shouldn't be needed by run the patch for all process creation but rather just for the parent one, because the patch updates the `execArgv` and environment variables.

So if, we run the patch and set after set `process.execPath` to a `node.exe`, we should be good. Alternatively we can have an `exe` with the patch which I think it was your initial idea.

**Comment by gregmagolan** [comment]
@googlebot I consent.

**Comment by alexeagle** [comment]
@googlebot I consent

**Pull request merged**